gammapy/maps/hpxmap.py

```diff
@@ -197,25 +225,29 @@ def make_hdu(self, **kwargs):
             The HDU extension name.
         hdu_bands : str
             The HDU extension name for BANDS table.
-        colbase : str
-            The prefix for column names
         sparse : bool
             Set INDXSCHM to SPARSE and sparsify the map by only
-            writing pixels with non-zero amplitude.
+            writing pixels with non-zero amplitude.            
+        conv : {'fgst-ccube','fgst-template','gadf',None}, optional        
+            FITS format convention.  If None this will be set to the
+            default convention of the map.
+
+        Returns
+        -------
+        hdu_out : `~astropy.io.fits.BinTableHDU` or `~astropy.io.fits.ImageHDU`
+            Output HDU containing map data.
         """
 
         from .hpxsparse import HpxSparseMap
 
-        conv = kwargs.get('conv', HpxConv.create('gadf'))
+        convname = self.geom.conv if conv is None else conv
+        conv = HpxConv.create(convname)
+        hduname = conv.hduname if hdu is None else hdu
+        hduname_bands = conv.bands_hdu if hdu_bands is None else hdu_bands
 
         data = self.data
         shape = data.shape
```

Comment:
Unused local variable `shape` -> remove?

Reply:
Yes this should be removed.